### PR TITLE
Feature/tao 4278 user friendly categories

### PR DESCRIPTION
**Pull request by no-chris**
https://oat-sa.atlassian.net/browse/TAO-4278

Companion PRs : 
https://github.com/oat-sa/tao-core/pull/1316
https://github.com/oat-sa/extension-tao-act/pull/543

### CODE REVIEW

File: models/classes/TestCategoryPreset.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA;
```
Comment (krampstudio) [inline models/classes/TestCategoryPreset.php]: Really 2016 ?

File: scripts/update/Updater.php [hunk 2]
```diff
@@ -1223,5 +1224,10 @@ public function update($initialVersion) {
             }
             $this->setVersion('9.2.0');
         }
+
+        if ($this->isVersion('9.2.0')) {
+            $this->getServiceManager()->register(TestCategoryPresetProvider::SERVICE_ID, new TestCategoryPresetProvider());
+//            $this->setVersion('9.3.0');
```
Comment (krampstudio) [inline scripts/update/Updater.php]: You have forgotten to uncomment

File: views/js/controller/creator/templates/itemref-props.tpl [hunk 3]
```diff
@@ -88,6 +72,29 @@
         </div>
     </div>
 
+    <div class="categories">
+        <div class="grid-row">
+            <div class="col-5">
+                <label for="category-custom">{{__ 'Categories'}}</label>
+            </div>
+            <div class="col-6">
+                <input type="text" title="custom categories" name="category-custom"/>
```
Comment (krampstudio) [inline views/js/controller/creator/templates/itemref-props.tpl]: The title needs also to be translated

**Comment by no-chris** [comment]
requested changes have been made

**Pull request merged**